### PR TITLE
Bumped dependencies and code for php 7.4 compatibility

### PR DESCRIPTION
**Pull request by Maaiins**
ref: #414  #432

### CODE REVIEW

**Comment by padams** [comment]
I wonder if we can eliminate the use of JSON.php. It was added way back when PHP didn't have built in JSON support. At a minimum i think we can add add some static methods to the owa_lib class such as  `owa_lib:json_encode` and owa_lib::json_decode that abstracts away the need for the rest of the code base to know about JSON.php. Thoughts?

**Comment by padams** [comment]
Great cleanup.

**Comment by Maaiins** [comment]
I think there are a few more things to improve. For example adding dependencies via composer. When doing this the cleanup of old deprecated libraries is also an option.
But I think there is no need to add static methods, you can just use built in functions like `json_encode()` or the Symfony Serializer component (For more advanced usage). I don't know which one is the right one at the moment.

**Comment by padams** [comment]
maybe try the built in php method first and let's see if that is compatible output.

**Comment by Maaiins** [comment]
I found out the `JSON.php` file is used in `owa_jsonView` and `owa_jsonResultsView` classes only. I searched for the usage of the classes but they are unused. Can that be correct? If that´s true simply remove the file and both classes.

I´ve done this for me, there was no error in admin or when tracking events.

**Comment by padams** [comment]
That is entirely possible. I did notice that the resultSetManager already uses the standard php json_encode method so it's likely that the refactoring started but was never completed.

File: owa_lib.php [hunk 1]
```diff
@@ -472,8 +472,8 @@ public static function fileInclusionFilter($str) {
 		if (strpos($str, '%00')) {
 			$str = '';
 		}
-		
-		if (strpos($str, null)) {
```
Comment (Maaiins) [inline owa_lib.php]: Check this line

**Pull request merged**